volkswagencarnet/vw_vehicle.py

```diff
@@ -161,19 +158,55 @@ async def update(self):
             await self.discover()
         if not self.deactivated:
             await asyncio.gather(
-                self.get_preheater(),
-                self.get_climater(),
-                self.get_trip_statistic(),
-                self.get_position(),
-                self.get_statusreport(),
-                self.get_charger(),
-                self.get_timerprogramming(),
-                return_exceptions=True,
+                # TODO: we don't check against capabilities currently, but this also doesn't seem to be neccesary
+                # to be checked if we should still do it for UI purposes
+                self.get_selectivestatus(
+                    [
+                        "access",
+                        "fuelStatus",
+                        "honkAndFlash",
+                        "userCapabilities",
+                        "vehicleLights",
+                        "vehicleHealthInspection",
+                        "measurements",
+                        "charging",
+                        "climatisation",
+                        "automation",
+                    ]
+                ),
+                self.get_vehicle(),
+                self.get_parkingposition()
+                #     self.get_preheater(),
+                #     self.get_climater(),
+                #     self.get_trip_statistic(),
+                #     self.get_position(),
+                #     self.get_statusreport(),
+                #     self.get_charger(),
+                #     self.get_timerprogramming(),
+                #     return_exceptions=True,
             )
         else:
             _LOGGER.info(f"Vehicle with VIN {self.vin} is deactivated.")
 
     # Data collection functions
+    async def get_selectivestatus(self, services):
+        """Fetch selective status for specified services."""
+        data = await self._connection.getSelectiveStatus(self.vin, services)
+        if data:
+            self._states.update(data)
+
+    async def get_vehicle(self):
+        """Fetch car masterdata."""
+        data = await self._connection.getVehicleData(self.vin)
+        if data:
+            self._states.update(data)
+
+    async def get_parkingposition(self):
+        """Fetch parking position."""
+        data = await self._connection.getParkingPosition(self.vin)
+        if data:
+            self._states.update(data)
```

Comment:
```suggestion
        if data := await self._connection.getParkingPosition(self.vin):
            self._states.update(data)
```
Just a perfect use for a walrus operator 😄

Reply:
Same argument again :) I tried to copy the original style as much as possible, and that's how it was done before.
My perspective on this is: rather have uniform code than optimizing every little detail once you stumble over it.

Reply:
Sorry, resolved the wrong convo